ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java

```diff
@@ -944,28 +941,42 @@ public void visit(Resource resource) {
     }
   }
 
-  private Promise<Optional<Resource>> findResource(final Path absolutePath, boolean quiet) {
-    return ps.getItem(absolutePath)
+  private Promise<Optional<Resource>> doFindResource(Path path) {
+    return ps.getTree(path.parent(), 1, true)
         .thenPromise(
-            itemReference -> {
-              final Resource resource = newResourceFrom(itemReference);
+            treeElement -> {
+              Resource resource = null;
 
-              store.register(resource);
+              for (TreeElement nodeElement : treeElement.getChildren()) {
+                ItemReference reference = nodeElement.getNode();
+                Resource tempResource = newResourceFrom(reference);
+                store.register(tempResource);
+
+                if (tempResource.isProject()) {
+                  inspectProject(tempResource.asProject());
+                }
 
-              if (resource.isProject()) {
-                inspectProject(resource.asProject());
+                if (tempResource.getLocation().equals(path)) {
+                  resource = tempResource;
+                }
               }
 
-              return promises.resolve(fromNullable(resource));
+              return promises.resolve(Optional.fromNullable(resource));
             })
-        .catchErrorPromise(
-            arg -> {
-              if (!quiet) {
-                throw new IllegalStateException(arg.getCause());
-              }
+        .catchErrorPromise(error -> promises.resolve(absent()));
+  }
 
-              return promises.resolve(absent());
-            });
+  private Promise<Optional<Resource>> findResource(final Path absolutePath, boolean quiet) {
+    String[] segments = absolutePath.segments();
+
+    Promise<Optional<Resource>> chain = promises.resolve(null);
+
+    for (int i = 0; i <= segments.length; i++) {
+      Path pathToRetrieve = absolutePath.removeLastSegments(segments.length - i);
+      chain = chain.thenPromise(__ -> doFindResource(pathToRetrieve));
```

Comment:
Try to find more informative name than "__"

Reply:
It's ignored value, `ignored` or `arg` name also looks not so good.